### ISSUE TITLE
3.2.6 Consistent Help wording is perhaps confusing, understanding uses "should" which may make it sound normative

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#2098]
The current normative wording for the SC

> For each web page within a set of web pages that provides one or more of the following ways of finding help, access to at least one form of help is included in the same relative order on each page:

may be slightly confusing, because on skim-reading it it appears to say that pages need to include at least one of these forms of help. Of course, careful re-reading shows that it's scoped only to pages that do have one or more of those forms of help, but perhaps the wording can be tweaked/turned around, if it's not too late?

though admittedly it's not completely non-trivial to come up with a better wording...maybe if the list can be - awkwardly - put in the middle like so?

> For each web page within a set of web pages that provides one or more of the following ways to find help
>
> * Human contact details;
> * Human contact mechanism;
> * Self-help option;
> * A fully automated contact mechanism
> 
> then access to at least one of these is included in the same relative order on each page

---

As originally noted in https://github.com/w3c/wcag/pull/1834 the understanding document for 3.2.6 also has a few instances of using the words `should` - while it's perhaps clear that it's not the normative `SHOULD`, and the understanding docs themselves are only informative, it may give the impression that it's nonetheless making normative claims/statements.

> At least one of the following ways to get help **should** be included on all websites:

it may be that the "While it would be best..." sentence could be moved before the bullet list, and combined with the spirit of this to basically say (heavily paraphrased): normatively this SC doesn't specify the types of help, but as a best practice, these are the sorts of things that would be good to have...

Further instances:

> The human contact details **should** enable the user to connect with the organization or the part of the organization that can assist with the content. For example, an online jobs / recruitment portal **should** provide a contact method for the team that supports the recruitment portal and not a catch-all for the entire company. Each layer of contact added prolongs the time before the user will receive help.

> Self-help options **should** go beyond allowing the user to search within the site. Contextual help is still recommended (See SC 3.3.5 for more information) but a self-help option **should** provide a single location that makes it easier for people with cognitive disabilities to understand what help is available without having to hunt for it.

> If a chatbot is provided, it **should** meet other WCAG Success Criteria for your conformance level. Chatbots which work more effectively for everyone, and particularly for people with cognitive disabilities **should**:

> If the human contact is not available during certain hours or certain days, information **should** be provided so the user can tell when it will be available.

**Comment by mbgower** [comment]
@mraccess77 
> People also might think that if you have a set a pages all of the pages in the set need to have help and that's not the case.

I thought at one point there was an expectation that at least one of these would occur on every page? For now, I have tried to redraft with your statement in mind.

@patrickhlauke, how about:

**> Within a set of web pages using any of the following methods of finding help, each method should occur in the same relative order on any page where it is used:**

Part of me wonders why we don't just adopt the 'established language of 3.2.3, which reads:
> Navigational mechanisms that are repeated on multiple Web pages within a set of Web pages occur in the same relative order each time they are repeated, unless a change is initiated by the user.

It would become:
**> Any of the following help mechanisms that are repeated on multiple Web pages within a set of Web pages occur in the same relative order each time they are repeated, unless a change is initiated by the user.**

**Comment by patrickhlauke** [comment]
> I thought at one point there was an expectation that at least one of these would occur on every page? For now, I have tried to redraft with your statement in mind.

Interesting...it depends on what "provides" refers to in

```
For each <a>web page</a> within a <a>set of web pages</a> that provides one or more of the following ways of finding help
```

Is it "for each web page ... that provides" or "for "... set of web pages that provides"

> **> Within a set of web pages using any of the following methods of finding help, each method should occur in the same relative order on any page where it is used:**
> 
> Part of me wonders why we don't just adopt the established language of 3.2.3, which reads:
> 
> > Navigational mechanisms that are repeated on multiple Web pages within a set of Web pages occur in the same relative order each time they are repeated, unless a change is initiated by the user.
> 
> It would become: **> Any of the following help mechanisms that are repeated on multiple Web pages within a set of Web pages occur in the same relative order each time they are repeated, unless a change is initiated by the user.**

I'm wondering if adding an "IF" at the start might help actually make it clearer on skim reading...e.g.

> If web pages within a set of web pages provide any of the following help mechanism, then access to these help mechanisms must occur in the same relative order on each web page in the set

or something. I think the front-loaded "IF" makes it much more immediate that no, this doesn't mean each page in the set MUST provide one or more of these

**Comment by mbgower** [comment]
@patrickhlauke Okay, I think the first thing we have to clafiry is the intent of the scope for this SC. @rachaelbradley can you please provide input -- both from original intent and from COGA's current understanding? It would really help get the normative language tuned up.
Key question:
1. is this supposed to require at least one form on every page of a set of pages where help is used on any of the pages? OR
2. is this supposed to require that any form of help used in a set of pages appears consistently on whichever of the pages in that set it is used on?

In example, the current Understanding document seems to imply that at least one from of help must be on every page, but even within these 2 paragraphs, there seems to be almost contradictory information

> The help mechanism, such as a contact phone number, may be provided directly on the page but it may also be a direct link to a different contact page. Regardless of which approach is used, the mechanism must be located in the same relative order on each page within the set of pages.

> The location in a smaller viewport may be different than in a larger viewport but the mechanism or link should be consistent across a set of web pages. The location should remain consistent both visually and programmatically.

**Comment by mraccess77** [comment]
Up until now it was written to say "2.	is this supposed to require that any form of help used in a set of pages appears consistently on whichever of the pages in that set it is used on?"

If that isn't the intent then we need to make that clear and make sure everyone is ok with that - because 1. above is a substantial change and not just editorial.

**Comment by alastc** [comment]
Adding a little ping-pong @mbgower and I did today, posting here for further thought:

> Any of the following help mechanisms that are repeated on multiple Web pages within a set of Web pages occur in the _same relative order_ to other page content, unless a change is initiated by the user:
> - Human contact details;
> - Human contact mechanism;
> - Self-help option;
> - A fully automated contact mechanism.
> 
> Note: For this SC, the same relative order can be thought of as how the content is ordered when the page is serialized. The visual position of a help mechanism is likely to be consistent across pages for the same page variation (e.g., CSS break-point). The user can cause a change, for example zoom or change orientation, which may trigger a different page variation.

**Comment by mbgower** [comment]
Note that "same relative order" is an existing defined term used in Consistent Navigation, which gets us a _little bit_ of a freebie by adding the idea of position

> same position relative to other items

https://www.w3.org/WAI/WCAG21/Understanding/consistent-navigation#dfn-same-relative-order

**Comment by mbgower** [comment]
@alastc with the tradition of the SC still being readable when the definition is substituted for the definition term, this may work better as:
> Any of the following help mechanisms that are repeated on multiple Web pages within a set of Web pages occur in the same relative order on the page, unless a change is initiated by the user:

**Comment by patrickhlauke** [comment]
I *think* this latest proposed rejigging works in alleviating most confusion. It's certainly far more straightforward than the current/original wording.

**Comment by alastc** [comment]
I've implemented that in #2145, keeping the "to other page content". 

Also, we have a competing PR with #2144, which is aiming to make clear that it only applies if:
- the help is provided elsewhere in the set of web pages, and
- the current page provides a help mechanism.

I think this version covers the first of those aspects, but not the second. Any ideas on that?
